theano/scan_module/scan_checkpoint.py

```diff
@@ -75,7 +75,7 @@ def outer_step(*args):
                                   n_steps=o_n_steps, allow_gc=True)
                                   
     # Keep only the last timestep of every output but keep all the updates
-    return results, updates
+    return results, updates # TODO is it a bug?
```

Comment:
@nouiz : I think the line was added so that using scan_with_checkpoints would be more like using standard scan (in the tests I did, I could just swap in one or the other and everything else would remain the same). Also, it made it possible to use scan_with_checkpoints to reduce memory usage when, for instance, you want every N-th output step (ex : steps 10, 20, 30, 40 and 50 out of 50 output steps). To be fair, I don't remember if this is something a user mentioned to me as being useful for his use case or this is just something I thought of myself so I don't know how used such a feature would be

Also, even if the user uses all that is inside 'results', he would still get some memory savings. 'results' at that line, only contains the N-th output steps (because the inner_step function only returns the last output steps).

Is my response clear enough or should we talk about this in person?